File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *
import string

#setrecursionlimit(10**9)
#input = stdin.readline

numbers = string.digits

def convert(n,base): #  n을 base진법
    
    a = n 
    ret = ""
    
    while a != 0:
        b = a // base
        c = a % base
        
        a = b
        ret = numbers[c] + ret
    
    return ret

def isPrime(n): # 소수 판별
    end = int(int(n)**0.5)+1
    
    if n == 1:
        return False
    
    if n == 2 or n == 3:
        return True
    
    
    for i in range(2,end+1):
        if n % i == 0:
            return False
    
    return True
    
    

def solution(n, k):
    answer = 0
    
    s = convert(n,k)
    
    s = s.split("0")
    
    for num in s:
        if num:
            if isPrime(int(num)):
                answer += 1

    return answer

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.swift
import Foundation


func isPrime(_ num:Int) -> Bool {
    
    if num == 1 {
        return false
    }
    var flag = true
    
    for i in 2..<Int(sqrt(Double(num)))+1{
        if num % i == 0 {
            flag = false
            break
        }
    }
    
    return flag
    
    
}

func solution(_ n:Int, _ k:Int) -> Int {
    
        
    return String(n,radix:k).split{$0 == "0"}.map{Int($0)!}.filter{isPrime($0)}.count
}